pkg/shortnames/shortnames.go

```diff
@@ -0,0 +1,474 @@
+package shortnames
+
+import (
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/containers/image/v5/docker/reference"
+	"github.com/containers/image/v5/pkg/sysregistriesv2"
+	"github.com/containers/image/v5/types"
+	"github.com/manifoldco/promptui"
+	"github.com/opencontainers/go-digest"
+	"github.com/pkg/errors"
+	"golang.org/x/crypto/ssh/terminal"
+)
+
+// IsShortName returns true if the specified input is a "short name".  A "short
+// name" refers to a container image without a fully-qualified reference, and
+// is hence missing a registry (or domain).  Names including a digest are not
+// short names.
+//
+// Examples:
+//  * short names: "image:tag", "library/fedora"
+//  * not short names: "quay.io/image", "localhost/image:tag",
+//                     "server.org:5000/lib/image", "image@sha256:..."
+func IsShortName(input string) bool {
```

Comment:
BTW neither https://github.com/containers/buildah/pull/2695 nor https://github.com/containers/podman/pull/7964 calls this, AFAICS. So maybe it can be dropped.